staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -111,6 +111,9 @@ func decodeManagedFields(encodedManagedFields []metav1.ManagedFieldsEntry) (mana
 	managed.fields = make(fieldpath.ManagedFields, len(encodedManagedFields))
 	managed.times = make(map[string]*metav1.Time, len(encodedManagedFields))
 	for _, encodedVersionedSet := range encodedManagedFields {
+		if encodedVersionedSet.FieldsType != "FieldsV1" {
+			return managedStruct{}, fmt.Errorf("missing fields type from %v", encodedVersionedSet)
```

Comment:
this is a confusing message to return if the value was non-empty and was something other than "FieldsV1"

Reply:
Yeah I haven't spent to much time on this message because, to be fair, it's mostly dropped on the floor (since on errors, we just reset managedFields to the one in liveObj)

Reply:
for the places where we do log or surface the error, maybe change to something like:
```golang
switch encodedVersionedSet.FieldsType {
  case "":
    return managedStruct{}, fmt.Errorf("missing fieldsType in managed fields entry %d", i)
  case "FieldsV1":
    // valid
  default:
    return managedStruct{}, fmt.Errorf("invalid fieldsType %q in managed fields entry %d", encodedVersionedSet.FieldsType, i)
}
```